### PR TITLE
fix: program indicator period boundaries are now validated [BETA-55]

### PR DESCRIPTION
**Pull request by gnespolino**

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/EnrollmentTimeFieldSqlRenderer.java
Comment (maikelarabori) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/EnrollmentTimeFieldSqlRenderer.java:137]: I believe would be better to use an ErrorCode, so clients can handle errors easier. ie.:

```
throw new IllegalQueryException( new ErrorMessage( E7136 ) );
```

**Review by maikelarabori — APPROVED** [review]
Looks good :+1:  - Just a minor comment @gnespolino.
Please, wait for the confirmation of the rule before merging it (maybe move it to draft until there?)

Thanks

**Review by d-bernat — APPROVED** [review]

**Comment by gnespolino** [comment]
this is not functionally correct, closing the PR

**Pull request closed without merging**
